dataflow_test.py

```diff
@@ -523,3 +523,71 @@ def test_count_filter_counts_correctly():
 
     assert result.n_passed == len( odd_numbers)
     assert result.n_failed == len(even_numbers)
+
+
+@parametrize("component",
+             (df.map   (lambda x: x)    ,
+              df.filter(lambda x: x > 0),
+              df.sink  (print)          ,
+              df.branch(df.sink(print)) ,
+              df.pipe  (df.map(abs))    ))
+def test_string_to_pick_ignores_components(component):
+    assert component is df.string_to_pick(component)
+
+
+def test_string_to_pick():
+
+    # string_to_pick creates a pipe component that picks
+    # an item from the namespace and pushes it through the pipe
+
+    the_source_elements = list(range(10))
+    the_source          = (dict(x=i) for i in the_source_elements)
+
+    result = []; the_sink = df.sink(result.append)
+    df.push(source = the_source,
+            pipe   = df.pipe(df.string_to_pick("x"), the_sink))
+
+    assert result == the_source_elements
+
+
+# string_to_pick is not meant to be called directly.
+# It's action is implicit when the first element of
+# a pipe is a string. This also works with forks
```

Comment:
Excactly: this agrees with my earlier comment about `string_to_pick` not being part of the API, and therefore having to place in these user-facing documentation tests.